pyasic/web/avalon.py

```diff
@@ -0,0 +1,84 @@
+# ------------------------------------------------------------------------------
+#  Copyright 2022 Upstream Data Inc                                            -
+#                                                                              -
+#  Licensed under the Apache License, Version 2.0 (the "License");             -
+#  you may not use this file except in compliance with the License.            -
+#  You may obtain a copy of the License at                                     -
+#                                                                              -
+#      http://www.apache.org/licenses/LICENSE-2.0                              -
+#                                                                              -
+#  Unless required by applicable law or agreed to in writing, software         -
+#  distributed under the License is distributed on an "AS IS" BASIS,           -
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.    -
+#  See the License for the specific language governing permissions and         -
+#  limitations under the License.                                              -
+# ------------------------------------------------------------------------------
+
+from __future__ import annotations
+
+import aiofiles
+import httpx
+import json
+from pathlib import Path
+from typing import Any, Optional
+
+from pyasic import settings
+from pyasic.web.base import BaseWebAPI
+
+
+class AvalonWebAPI(BaseWebAPI):
+    def __init__(self, ip: str) -> None:
+        super().__init__(ip)
+        self.username = "admin"
+        self.pwd = settings.get("default_avalon_web_password", "admin")
+        self.port = 80
+        self.token = None
+
+    async def auth(self) -> None:
+        """Authenticate and get the token. Implement the actual authentication logic here."""
+        self.token = "Success"
+
+    async def send_command(
```

Comment:
This function should be generic, and a function like `upgrade` should just pass the file data in the params.